integration-tests/src/test/java/oracle/weblogic/kubernetes/utils/CommonTestUtils.java

```diff
@@ -4023,4 +4026,79 @@ public static void createIngressAndRetryIfFail(int maxRetries,
       }
     }
   }
+
+  /**
+   * Build image with unique name, create corresponding docker secret and push to registry.
+   *
+   * @param dockerFileDir directory where dockerfile is located
+   * @param baseImageName base image name
+   * @param namespace image namespace
+   * @param secretName docker secretname for image
+   * @param extraDockerArgs user specified extra docker args
+   * @return image name
+   */
+  public static String createPushImage(String dockerFileDir, String baseImageName,
+                                       String namespace, String secretName,
+                                       String extraDockerArgs) throws ApiException {
+    // create unique image name with date
+    final String imageTag = TestUtils.getDateAndTimeStamp();
+    // Add repository name in image name for Jenkins runs
+    final String imageName = DOMAIN_IMAGES_REPO + baseImageName;
+
+    final String image = imageName + ":" + imageTag;
+    LoggingFacade logger = getLogger();
+    //build image
+    assertTrue(Docker.createImage(dockerFileDir, image, extraDockerArgs), "Failed to create image " + baseImageName);
+    logger.info("image is created with name {0}", image);
+    if (!new Namespace().exists(namespace)) {
```

Comment:
Isn't this a static method, do we need "new"?

Reply:
done